src/sv_commands.cpp

```diff
@@ -1061,6 +1061,33 @@ void SERVERCOMMANDS_PlayerSay( ULONG ulPlayer, const char *pszString, ULONG ulMo
 	}
 }
 
+//*****************************************************************************
+//
+void SERVERCOMMANDS_PrivateSay( ULONG ulSender, ULONG ulReceiver, const char *pszString, bool bForbidChatToPlayers, ULONG ulPlayerExtra, ServerCommandFlags flags )
+{
+	if ( ulSender != MAXPLAYERS && PLAYER_IsValidPlayer( ulSender ) == false )
+		return;
+
+	ServerCommands::PlayerSay command;
+	command.SetMessage( pszString );
+
+	// [AK] First send the command to the player receiving the message.
+	if (( ulReceiver != MAXPLAYERS ) && ( !bForbidChatToPlayers || players[ulReceiver].bSpectating ))
+	{
+		command.SetPlayerNumber( ulSender );
+		command.SetMode( CHATMODE_PRIVATE_RECEIVE );
+		command.sendCommandToClients( ulReceiver, SVCF_ONLYTHISCLIENT );
+	}
+
+	// [AK] Next, send the command back to the player who sent the message.
+	if ( ulSender != MAXPLAYERS )
```

Comment:
This will happen even if the message was not send due to `(( ulReceiver != MAXPLAYERS ) && ( !bForbidChatToPlayers || players[ulReceiver].bSpectating ))` being false. I think in that case, the sender should receive an error.